consensus_test.go

```diff
@@ -522,12 +902,34 @@ func newMockPbft(t *testing.T, accounts []string, account string) *mockPbft {
 	return m
 }
 
-func (i *mockPbft) Close() {
-	i.cancelFn()
+func getDefaultLoggerOutput() io.Writer {
+	if os.Getenv("SILENT") == "true" {
+		return ioutil.Discard
+	}
+	return os.Stdout
+}
+
+func newMockBackend(
+	validatorIds []string,
+	mockPbft *mockPbft,
+	buildProposal buildProposalDelegate,
```

Comment:
These callbacks are cumbersome to read and you have to use `nil` in much of the definitions of `newMockBackend`. We might use this pattern instead:

```
func (m *mockBackend) WithBuildProposal(buildProposal buildProposalDelegate) *mockBackend {
    m.buildProposal  = buildProposal
}

func TestXX(t *testing.T) {
   m := newmockBackend([]string{}, ..).WithBuildProposal(func...)
}
```